### PR TITLE
v1: cancel deployment deletion

### PR DESCRIPTION
**Pull request by kopiczko**
The problem is the cert secrets can be already deleted.

### CODE REVIEW

**Comment by kopiczko** [comment]
Need LGTM <3

File: service/controller/v1/resource/deployment/resource.go [hunk 1]
```diff
@@ -39,19 +38,15 @@ type Config struct {
...
 
 	azure           setting.Azure
 	azureConfig     client.AzureConfig
 	templateVersion string
 }
 
 func New(config Config) (*Resource, error) {
-	if config.CertsSearcher == nil {
-		return nil, microerror.Maskf(invalidConfigError, "%T.CertsSearcher must not be empty", config)
-	}
```
Comment (xh3b4sd) [inline service/controller/v1/resource/deployment/resource.go]: Why is the certs searcher now gone?

File: service/controller/v1/resource/deployment/resource.go [hunk 2]
```diff
@@ -90,6 +84,13 @@ func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interf
 		return nil, microerror.Mask(err)
 	}
 
+	// Deleting the resource group will take care about cleaning
+	// deployments.
```
Comment (xh3b4sd) [inline service/controller/v1/resource/deployment/resource.go]: A bit more information on why we cancel would be cool. IIRC there was a nice comment already in your previous commit containing these information.

**Comment by kopiczko** [comment]
I'm going with this now. e2e tests takes time and I need to run a few more. If comments need refining let's discuss further.

**Comment by xh3b4sd** [comment]
All good from my side. Thank you.

**Pull request merged**